test/extensions/filters/http/set_filter_state/integration_test.cc

```diff
@@ -66,6 +67,44 @@ class SetMetadataIntegrationTest : public testing::Test {
     EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter->decodeHeaders(headers_, true));
   }
 
+  void runPerRouteFilter(const std::string& listener_yaml_config,
+                         const std::string& per_route_yaml_config) {
+    Server::GenericFactoryContextImpl generic_context(context_);
+
+    envoy::extensions::filters::http::set_filter_state::v3::Config listener_proto_config;
+    TestUtility::loadFromYaml(listener_yaml_config, listener_proto_config);
+    auto listener_config = std::make_shared<Filters::Common::SetFilterState::Config>(
+        listener_proto_config.on_request_headers(), StreamInfo::FilterState::LifeSpan::FilterChain,
+        generic_context);
+
+    envoy::extensions::filters::http::set_filter_state::v3::Config route_proto_config;
+    TestUtility::loadFromYaml(per_route_yaml_config, route_proto_config);
+    Filters::Common::SetFilterState::Config route_config(
+        route_proto_config.on_request_headers(), StreamInfo::FilterState::LifeSpan::FilterChain,
+        generic_context);
+
+    NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+    EXPECT_CALL(*decoder_callbacks.route_, mostSpecificPerFilterConfig(_))
+        .WillOnce(Return(&route_config));
+
+    auto filter = std::make_shared<SetFilterState>(listener_config);
+    filter->setDecoderFilterCallbacks(decoder_callbacks);
+    EXPECT_CALL(decoder_callbacks, streamInfo()).WillRepeatedly(ReturnRef(info_));
+    EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter->decodeHeaders(headers_, true));
+
+    // Test the factory method.
+    {
+      NiceMock<Server::Configuration::MockServerFactoryContext> context;
+      SetFilterStateConfig factory;
+      Router::RouteSpecificFilterConfigConstSharedPtr route_config =
+          factory
+              .createRouteSpecificFilterConfig(route_proto_config, context,
+                                               ProtobufMessage::getNullValidationVisitor())
+              .value();
+      EXPECT_TRUE(route_config.get());
+    }
```

Comment:
This should be part of config_test, I think, if there is one.

Reply:
I don't think there is one, just this test. I was following how the existing filter level test is written